lib/ansible/plugins/inventory/__init__.py

```diff
@@ -449,6 +450,19 @@ def _add_host_to_keyed_groups(self, keys, variables, host, strict=False, fetch_h
                             result_gname = self.inventory.add_group(gname)
                             self.inventory.add_host(host, result_gname)
 
+                            # Add `ansible_keyed_group_name` to the variables that we can use to template
+                            templar_variables = combine_vars(variables, {'ansible_keyed_group_name': result_gname})
+                            self.templar.available_variables = templar_variables
+                            # Template the group_priority. There are not many variables available at this point,
+                            # But what we have should be able to be used here.
+                            group_priority = self.templar.template(keyed.get('priority', None))
+                            if group_priority is not None:
+                                result_group = self.inventory.groups.get(result_gname)
+                                try:
+                                    result_group.set_priority(int(group_priority))
+                                except ValueError as e:
+                                    raise AnsibleParserError("Invalid group priority given: %s" % to_native(e))
+
```

Comment:
Maybe don't do any extra work in the common case where it's not specified, get rid of Py2-isms, leaky templar changes, walrus, move the error handling into the place that's FIXME in `group.set_priority` so we're not coercing ints and error checking in multiple places:

```suggestion
                            if (priority := keyed.get('priority')) is not None:
                                with self.templar.set_temporary_context(available_variables=variables | {'ansible_keyed_group_name': result_gname}):
                                    if (priority := self.templar.template(priority)) is not None:
                                        try:
                                            self.inventory.groups.get(result_gname).set_priority(priority)
                                        except ValueError as ve:
                                            raise AnsibleParserError(f"Invalid group priority given: {ve}")
```